### PR TITLE
FEAT: Delete post by postId

### PR DESCRIPTION
**Pull request by alpha037**
- [x] Delete post by postId
delete-post-by-id

- [x] Post cannot be deleted if it's already deleted or if there's no post with that id
cannot-delete-post

### CODE REVIEW

File: src/posts/posts.controller.ts [hunk 1]
```diff
@@ -66,10 +66,12 @@ export class PostsController {
 
   @ApiBearerAuth()
   @UseGuards(RequiredAuthGuard)
-  @Delete('/:postid')
-  async deletePost(@Param('postid') postid: string): Promise<string> {
+  @Delete('/:postId')
+  async deletePost(@Param('postId') postId: string): Promise<string> {
     // TODO
-    return `delete postid = ${postid}`;
+    return (await this.postsService.deletePost(postId))
```
Comment (championswimmer) [inline src/posts/posts.controller.ts]: Since this is a REST API, best if all controllers always only return json objects and not bare strings anywhere

**Review by championswimmer — CHANGES_REQUESTED** [review]

File: src/posts/posts.controller.ts (hunk 1 above)
Comment (alpha037) [reply src/posts/posts.controller.ts]: Yeah, that totally makes sense. I've updated the `deletePost` controller. Let me know if you need any changes to that. :)

***After Update***

- [x] Returns true on successful delete
delete-post-by-id

- [x] Returns false if it's already deleted or if there's no post with that id
cannot-delete-post

**Comment by championswimmer** [comment]
We should also use a status code of `202` for `Accepted` when performing delete requests

**Comment by alpha037** [comment]
Sure, I'll change it.

**Pull request merged**